core/src/ast/idents.rs

```diff
@@ -10,8 +10,8 @@ pub struct Ident(Cow<'static, str>);
 
 impl Ident {
     /// Validate a string
-    fn validate(string: &str) {
-        let _ = syn::parse_str::<syn::Ident>(string).expect("Invalid identifier");
+    fn validate(string: &str) -> syn::Result<()> {
```

Comment:
oh I was imagining that this would be `try_new()` and return a `Result<Ident>` but i guess this is fine